src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx

```diff
@@ -32,7 +44,107 @@ import ArtifactTagRuntimeField from '../ArtifactSourceRuntimeFields/ArtifactTagR
 import css from '../../K8sServiceSpec.module.scss'
 
 interface ArtifactoryRenderContent extends ArtifactSourceRenderProps {
-  isTagsSelectionDisabled: (data: ArtifactSourceRenderProps) => boolean
+  isTagsSelectionDisabled: (data: ArtifactSourceRenderProps, isServerlessDeploymentTypeSelected: boolean) => boolean
+}
+
+interface TagFieldsProps extends ArtifactoryRenderContent {
+  template: ServiceSpec
+  stageIdentifier: string
+  path?: string
+  allowableTypes: MultiTypeInputType[]
+  fromTrigger?: boolean
+  artifact?: PrimaryArtifact | SidecarArtifact
+  selectedDeploymentType: ServiceDeploymentType
+  isSidecar?: boolean
+  artifactPath?: string
+  isTagsSelectionDisabled: (data: ArtifactSourceRenderProps, isServerlessDeploymentTypeSelected: boolean) => boolean
+  fetchingTags: boolean
+  artifactoryTagsData: ResponseArtifactoryResponseDTO | null
+  fetchTagsError: GetDataError<Failure | Error> | null
+  fetchTags: (props?: any) => Promise<void> | undefined
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
  fetchTags: (props?: unknown) => Promise<void> | undefined
```